src/dbus.c

```diff
@@ -283,6 +288,50 @@ static void dbus_cb_dunst_NotificationShow(GDBusConnection *connection,
         g_dbus_connection_flush(connection, NULL, NULL, NULL);
 }
 
+static void dbus_cb_dunst_NotificationListAll(GDBusConnection *connection,
+                                           const gchar *sender,
+                                           GVariant *parameters,
+                                           GDBusMethodInvocation *invocation)
+{
+        LOG_D("CMD: Listing all notifications from history");
+
+        GVariant *answer = NULL;
+        GVariantBuilder *builder;
+
+        builder = g_variant_builder_new(G_VARIANT_TYPE("aas"));
+
+        GList *notification_list = queues_get_history();
+
+        // reverse chronological list
+        for(int i = queues_length_history(); i > 0; i--) {
+                GVariantBuilder *notif_builder;
+                struct notification *n;
+                n = g_list_nth_data(notification_list, i-1);
+
+                // create sub-list per notification
+                notif_builder = g_variant_builder_new(G_VARIANT_TYPE("as"));
+
+                g_variant_builder_add(notif_builder, "s", n->body);
```

Comment:
Null strings aren't allowed here it seems. I get the output `CRITICAL: g_variant_new_string: assertion 'string != NULL' failed`

Reply:
so im deciding whether i should throw an dbus error or just return a literal empty string. probably the former

Reply:
I think you should either return an empty string or a string that signals it's null like `(null)`. It's not an error when variables in  a notification are set to NULL.

Reply:
id probably go for the empty string since a falsely crafted notification could also have "(null)"

Reply:
They could also have `""` :p. But I'd be fine with either.